changelog/issue-5079.md

```diff
@@ -0,0 +1,7 @@
+audience: deployers
+level: patch
+reference: issue 5079
+---
+The new `app.authRateLimitMaxRequests` configuration value allows setting the maximum number of web server requests per minute to help mitigate DoS attacks. Default is 5. Set to 0 to disable rate limiting.
+
+Another configuration value, `app.numberOfProxies` was added to set the number of proxies between the user and the server to prevent the rate limiter from acting like a global one. Default is 0. A value of 0 means that the first untrusted address would be req.socket.remoteAddress, i.e. there is no reverse proxy.
```

Comment:
Just to clarify, that means hitting the docker container itself directly right? Might be worth changing the default to 1 in helm charts then since you'll always have an ingress in front AFAIK?

Reply:
Interesting idea. If the ingress acts as a proxy, then we would see all traffic from one IP, and be limiting ourselves (default case) to 5 logins a minute. This may be a reasonable default - I have no idea what the statistics are for manual login on either deployment.

I'll see if I can figure out rates and the ingress proxy question from logs.

Reply:
I'd really love to be able to pinpoint a max value to use here for number of proxies.

On the [express-rate-limit docs](https://www.npmjs.com/package/express-rate-limit#troubleshooting-proxy-issues) they mention a way to get the number of proxies. Wasn't sure if this would be possible to try or not.

Reply:
Sorry, I wasn't able to check the logs to see if I could figure out the proxy levels.